File: Maximize Product.cpp
#include<iostream>
#include<vector>

using namespace std;

#define MOD 1000000007

bool possible = true;
vector<long long> vec;

void solve(long long n, long long k){
    vec.clear();
    //cout << "solving\n";
    vec.resize(k);
    for(int i = 0; i < k-1; i++){
        vec[i] = (long long)(i+1);
    }
    vec[k-1] = n - (k*(k-1))/2;

    if(vec[k-1] < k){
        possible = false;
        return;
    }
    if(k==1) return;
    long long inc = (vec[k-1] - vec[k-2] - k + 1)/k; 
    for(int i = 0; i < k-1; i++) vec[i]+=inc;
    vec[k-1] -= (inc*(k-1));
    long long cnt = vec[k-1] - vec[k-2] - 2;
    if(cnt > 0){
        for(int i = 0; i < cnt; i++){
            vec[k-2 - i]++;
        }
        vec[k-1] -= cnt;
    }
}

int main(){
    long long t, n, k;
    cin >> t;
    while(t--){
        cin >> n >> k;
        possible = true;
        solve(n, k);
        if(possible){
            long long res = 1;
            for(long long i: vec){
                //cout << i << " ";
                //res = (res * (i*i - i)) % MOD;
                res = (res*i)%MOD;
                res = (res*(i-1))%MOD;
            }
            //cout << endl;
            cout << res << endl;
        }
        else cout << "-1\n";
    }
}

File: Maximize Product.py
for _ in range(int(input())):
    MOD = 1000000007
    n, k = (int(x) for x in input().split())
    if k==1:
        print((int)(n*n - n) % MOD)
        continue
    possible = True
    a = [i for i in range(1, k)]
    a.append(n - (k*(k-1))/2);
    if a[k-1] < k:
        print(-1)
        continue
    inc = (a[k-1] - a[k-2] - k + 1)//k;
    for i in range(k-1):
        a[i]+=inc
    a[k-1] -= (inc*(k-1))
    cnt = (int)(a[k-1] - a[k-2] - 2)
    if cnt > 0:
        for i in range(cnt):
            a[k-2-i]+=1
        a[k-1]-=cnt

    lim = k-2
    while True:
        r = -1
        for i in range(lim, 1, -1):
            if a[i] != a[i+1] -1:
                r = i
                break
        if r==-1:
            break
        x = 0
        y = r
        while x < y:
            a[x]-=1
            a[y]+=1
            x+=1
            y-=1
        lim = r
            
    res = 1
    for i in a:
        res= (res*(i*i - i))%MOD
    print((int)(res)%MOD)
